scripts/Muon/GUI/ElementalAnalysis2/ea_group.py

```diff
@@ -47,13 +50,18 @@ def __del__(self):
                     remove_ws_if_present(table)
                 remove_ws_if_present(self.get_matches_table())
 
-        except Exception as e:
+        except Exception as error:
             """
                 If ADS is deleted before group is deleted, boost.python.ArgumentError is raised and
                 boost.python.ArgumentError are not catchable
             """
-            if "Python argument types in" not in str(e):
-                raise e
+            if "Python argument types in" not in str(error):
```

Comment:
we are best of catching the type of exceptions. e.g. ValueError or RunTimeError

Reply:
The Boost.python.ArgumentError raised when the ads is deleted is not catchable so it had to be done like this and all other exceptions will raise  a warning box telling users of an error